web/src/lib/components/shared-components/__test__/number-range-input.spec.ts

```diff
@@ -6,15 +6,22 @@ describe('NumberRangeInput component', () => {
   const user = userEvent.setup();
   let sut: RenderResult<NumberRangeInput>;
   let input: HTMLInputElement;
+  const props: { id: string; min: number; max: number; onInput: () => void; value: number | null } = $state({
+    id: '',
+    min: -90,
+    max: 90,
+    onInput: () => {},
+    value: null,
+  });
 
   beforeEach(() => {
-    sut = render(NumberRangeInput, { id: '', min: -90, max: 90, onInput: () => {} });
+    sut = render(NumberRangeInput, props);
     input = sut.getByRole('spinbutton') as HTMLInputElement;
   });
 
   it('updates value', async () => {
     expect(input.value).toBe('');
-    await act(() => sut.component.$set({ value: 10 }));
+    await act(() => (props.value = 10));
```

Comment:
Passing a `$state` as props and modifying that feels a bit messy. How about using `sut.rerender({ value: 10 })` instead?

Reply:
It felt messy to me as well, but it's the official way of how you're supposed to do this. https://svelte.dev/docs/svelte/v5-migration-guide#Components-are-no-longer-classes

Reply:
That's only the case when using `mount` or `hydrate` directly. The testing library internally uses $state for the props, so we just need to call the `rerender` function.